libraries/Microsoft.Bot.Builder.Ai/QnAMakerMiddleware.cs

```diff
@@ -32,18 +32,34 @@ public QnAMakerMiddleware(QnAMakerMiddlewareOptions options, HttpClient httpClie
                     if (results.Any())
                     {
                         if (!string.IsNullOrEmpty(_options.DefaultAnswerPrefixMessage))
-                            await context.SendActivity(_options.DefaultAnswerPrefixMessage);
-
-                        await context.SendActivity(results.First().Answer);
-
-                        if (_options.EndActivityRoutingOnAnswer)
-                            //Question is answered, don't keep routing
-                            return;
+                        {
+                            if (_isLastMiddleware)
+                                await context.SendActivity(_options.DefaultAnswerPrefixMessage);
+                            else
+                                messageActivity.Text = _options.DefaultAnswerPrefixMessage;
+                        }
+                        if (_isLastMiddleware)
```

Comment:
Unfortunalty, this whole "IsLast()" pattern probalby isn't going to be doable. The change is bigger than it seems at first glance.